### PR TITLE
drivers: scmi-msg: correct voltage domain protocol version & clock rates enumeration

### PR DESCRIPTION
**Pull request by etienne-lms**
Fix the version ID of the implemented SCMI voltage domain protocol that is v2.0 (ID 0x20000), not v3.0 (ID 0x30000).
(Fixes: 006d89b8f49f ("drivers: scmi-msg: add SCMI Voltage Domain protocol"))

Fix value of remaining clocks to describe on SCMI clock protocol message CLOCK_DESCRIBE_RATES that does not take into account the number of returned clock in the response.
(Fixes: 90252e2a52c7 ("drivers: scmi-msg: clock adapts to output buffer size"))

### CODE REVIEW

File: core/drivers/scmi-msg/clock.c [hunk 1]
```diff
@@ -321,7 +321,7 @@ static void scmi_clock_describe_rates(struct scmi_msg *msg)
 		}
 
 		out_count = rate_index - in_args->rate_index;
-		remaining = nb_rates - in_args->rate_index;
+		remaining = nb_rates - in_args->rate_index - out_count;
```
Comment (jforissier) [inline core/drivers/scmi-msg/clock.c:324]: `nb_rates - rate_index` would be better, no?

**Review by jforissier — COMMENTED** [review]
For "drivers: scmi-msg: correct voltage domain protocol version":
`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

File: core/drivers/scmi-msg/clock.c (hunk 1 above)
Comment (GseoC) [reply core/drivers/scmi-msg/clock.c:324]: I would have said `nb_rates - out_count` as out_count is the number of entries filled in the buffer. Maybe I'm missing something.

Comment (etienne-lms) [reply core/drivers/scmi-msg/clock.c:324]: > nb_rates - rate_index would be better, no?

Indeed simpler.

> I would have said `nb_rates - out_count` as out_count is the number of entries filled in the buffer. Maybe I'm missing something.

`nb_rates` gives the overall number of supported rates. `out_count` gives the number of rates filled in this very response message but does not count the rates skipped, from index 0 to index `in_args->rate_index - 1`.

**Review by etienne-lms — COMMENTED** [review]
I'll update the change.

**Comment by jforissier** [comment]
`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Review by GseoC — APPROVED** [review]

**Comment by etienne-lms** [comment]
Fixup commit squashed and review tag applied.
@GseoC, do you want to provide a review tag?

pkcs11_1007 test sporadic failure faced during Ci test for this P-R but it's unrelated to this change (I still don't understand exactly what happens there, still tracking through https://github.com/OP-TEE/optee_os/issues/6952).

**Comment by GseoC** [comment]
Sure, `Acked-by: Gatien Chevallier <gatien.chevallier@foss.st.com>`

**Comment by etienne-lms** [comment]
Review tags applied. Thanks.

**Pull request merged**